### PR TITLE
Upgrade to node 18

### PR DESCRIPTION
**Pull request by maliroteh-sf**
Node 16 will hit end-of-life on 2023-09-11 so we need to move to v18

### CODE REVIEW

File: force-app/main/default/lwc/accountRelatedContacts/accountRelatedContacts.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { graphql, gql } from "lightning/uiGraphQLApi";
 
 // eslint-disable-next-line @salesforce/lwc-graph-analyzer/no-unresolved-parent-class-reference
 export default class AccountRelatedContacts extends NavigationMixin(
-  LightningElement
+  LightningElement,
```
Comment (dbreese) [inline force-app/main/default/lwc/accountRelatedContacts/accountRelatedContacts.js:8]: Interesting that this is a node 18/20 need? This seems odd to me.

**Review by dbreese — APPROVED** [review]

File: sfdx-project.json [hunk 2]
```diff
@@ -7,5 +7,5 @@
   ],
   "namespace": "",
   "sfdcLoginUrl": "https://login.salesforce.com",
-  "sourceApiVersion": "58.0"
+  "sourceApiVersion": "59.0"
```
Comment (khawkins) [inline sfdx-project.json:10]: API 59 hasn't hit much of production yet—sandboxes only. Depending on when we want to pull the trigger on merging this PR, that's a consideration.

Comment (khawkins) [reply sfdx-project.json:10]: Unless there's something specific about Node 18 accounting for this change, I'd recommend reverting it for the time being.

Comment (maliroteh-sf) [reply sfdx-project.json:10]: @khawkins after upgrading `@salesforce/sfdx-lwc-jest` to the latest, executing `npm run test` was causing this error to be printed out (even though the tests were still running & passing(:
```error Invalid sourceApiVersion found in sfdx-project.json. Expected 59.0, found 58.0```

This is why I updated the API version. Do you think I should downgrade that dependency or should we leave it at the latest and wait for API 59 to hit production before merging?

Comment (khawkins) [reply sfdx-project.json:10]: If downgrading the dependency doesn't have its own set of adverse impacts, I'd go that route. Full rollout of the next major release version (and thus API 59 support) is still a ways out, and notably past the cutoff for the Node 16 EOL.

**Review by khawkins — APPROVED** [review]

**Pull request merged**